src/CPK.ts

```diff
@@ -34,173 +30,203 @@ class CPK {
     const wc = provider && (provider.wc || (provider.connection && provider.connection.wc))
     if (wc && wc.peerMeta && wc.peerMeta.name && wc.peerMeta.name.startsWith('Gnosis Safe')) {
-      this.isConnectedToSafe = true
+      this.#isConnectedToSafe = true
     }
 
-    this.multiSend = this.ethLibAdapter.getContract(multiSendAbi, network.multiSendAddress)
+    this.#multiSend = this.#ethLibAdapter.getContract(multiSendAbi, network.multiSendAddress)
 
-    if (this.isConnectedToSafe) {
-      this.contract = this.ethLibAdapter.getContract(safeAbi, ownerAccount)
+    if (this.isSafeApp() || this.#isConnectedToSafe) {
+      this.#contract = this.#ethLibAdapter.getContract(safeAbi, ownerAccount)
     } else {
-      this.proxyFactory = this.ethLibAdapter.getContract(cpkFactoryAbi, network.proxyFactoryAddress)
+      this.#proxyFactory = this.#ethLibAdapter.getContract(
+        cpkFactoryAbi,
+        network.proxyFactoryAddress
+      )
 
-      const salt = this.ethLibAdapter.keccak256(
-        this.ethLibAdapter.abiEncode(['address', 'uint256'], [ownerAccount, predeterminedSaltNonce])
+      const salt = this.#ethLibAdapter.keccak256(
+        this.#ethLibAdapter.abiEncode(
+          ['address', 'uint256'],
+          [ownerAccount, predeterminedSaltNonce]
+        )
       )
-      const initCode = this.ethLibAdapter.abiEncodePacked(
-        { type: 'bytes', value: await this.proxyFactory.call('proxyCreationCode', []) },
+      const initCode = this.#ethLibAdapter.abiEncodePacked(
+        { type: 'bytes', value: await this.#proxyFactory.call('proxyCreationCode', []) },
         {
           type: 'bytes',
-          value: this.ethLibAdapter.abiEncode(['address'], [network.masterCopyAddress])
+          value: this.#ethLibAdapter.abiEncode(['address'], [network.masterCopyAddress])
         }
       )
-      const proxyAddress = this.ethLibAdapter.calcCreate2Address(
-        this.proxyFactory.address,
+      const proxyAddress = this.#ethLibAdapter.calcCreate2Address(
+        this.#proxyFactory.address,
         salt,
         initCode
       )
 
-      this.contract = this.ethLibAdapter.getContract(safeAbi, proxyAddress)
+      this.#contract = this.#ethLibAdapter.getContract(safeAbi, proxyAddress)
     }
   }
 
   isSafeApp(): boolean {
-    return !!this.safeAppInfo
+    return this.#safeAppsSdkConnector.isSafeApp()
   }
 
-  async getOwnerAccount(): Promise<Address> {
-    if (this.ownerAccount) {
-      return this.ownerAccount
+  async getOwnerAccount(): Promise<Address | undefined> {
+    if (this.#ownerAccount) {
+      return this.#ownerAccount
+    }
+    if (this.isSafeApp()) {
+      return this.#safeAppsSdkConnector.safeAppInfo?.safeAddress
     }
-    if (!this.ethLibAdapter) {
+    if (!this.#ethLibAdapter) {
       throw new Error('CPK uninitialized ethLibAdapter')
     }
-    return this.ethLibAdapter.getAccount()
+    return this.#ethLibAdapter?.getAccount()
+  }
+
+  get safeAppsSdkConnector() {
```

Comment:
How much of an interface do we want to make public? This `safeAppsSdkConnector` and the corresponding `transactionManager` does very similar things... I'm not sure why they are different objects. If you want to merge the connector into the manager in the future, I'd suggest removing this accessor so that this is not part of the public API.